doc/architecture/architecture.rst

```diff
@@ -1,24 +1,252 @@
+`meta-security-isafw`_       Image Security Analysis framework gives
+                             offline tooling to analyze images
+`meta-soletta`_              `Soletta Project`_
+`meta-swupd`_                Software update tooling
+`meta-intel`_                BSP layer for common IA platforms
+`meta-yocto`_                BSP layer for BeagleBone Black
+`meta-edison-bsp`_           BSP layer for Intel Edison
+============================ =======================================
+
+The Ostro OS distro configuration file (:file:`meta-ostro/conf/distro/ostro.conf`) defines
+the global configuration of the OS. For example, it determines which components are being
+enabled/built by default (via ``DISTRO_FEATURES``, ``PNBLACKLIST``, and ``PNWHITELIST`` settings).
+
+Ostro OS Development Workflows
+==============================
+
+.. _`Yocto Project extensible SDK`: http://www.yoctoproject.org/docs/2.1/sdk-manual/sdk-manual.html#sdk-extensible
+
+Ostro OS enables many options how to do hacking and/or IoT product development.
+
+#. **Quick Prototyping**: Using the pre-built development images found in `Ostro Project download server`_ it's easy to get started with any of the :ref:`platforms`. Some of the image configurations also provide an environment to do quick prototyping with programming tools discussed in 'Programming Interfaces' above.
+#. **Image Customization**: If the pre-built development images do not provide the desired content or configuration, images can easily be customized to suit the needs. Alternatively, :ref:`software-update` could be used to extend the image content by installing additional bundles.
+#. **Adding 3rd party content**: For device makers it's often necessary to bring in additional 3rd party libraries and product IoT applications while still keeping the base OS unchanged. `Yocto Project extensible SDK`_ (discussed below) built for Ostro OS tries to make that process easy.
+
+Extensible SDK
+______________
+
+Using the `Yocto Project extensible SDK`_ built for Ostro OS makes it easy to add new applications and libraries to an image, modify the source for an existing component, test changes on the target hardware. Each Ostro OS build found in `Ostro Project download server`_ has an extensible SDK installer that is quick to download and install on user's development PC.
+
+The ``devtool`` is the main tool when using extensible SDK. With ``devtool`` user can add new recipes/content, e.g., from Node.JS NPM registry and have them installed in the image or modify kernel configuration options. To build the new components and new images, the Ostro OS pre-compiled SSTATE cache (including the cross-development toolchain) is used (as long as it's valid).
+
+Ostro OS Technical Documentation
+================================
+
+The following documents give more technical details about specific Ostro
+OS components and their usage.
 
 .. toctree::
    :maxdepth: 1
 
-   disk-layout
-   efi-boot
    software-update
```

Comment:
@mythi  can you add     software-update-server     after this line and resubmit (this will get the updated doc that's already been merged into the table-of-contents   @gvancuts